### ISSUE TITLE
Font.default does not reset when recompiling the library

### ISSUE THREAD

**Issue by jamshark70** [supercollider/supercollider#3685]
Start with a fresh sclang session ("reboot interpreter"). Then:

```
// This displays with the normal default font
StaticText(nil, Rect(800, 200, 100, 20)).string_("test").front;

// This displays with serifs
Font.default = Font("Times New Roman", 12);
StaticText(nil, Rect(800, 200, 100, 20)).string_("test").front;

thisProcess.recompile;

// This *still* displays with serifs
StaticText(nil, Rect(800, 200, 100, 20)).string_("test").front;
```

Expected behavior: Recompiling the class library should generally serve as a "reset" -- users assume any settings that were changed will go back to normal. (`Process.tailCallOptimize` is one of the few exceptions, but that's because it can be set only by an sclang instruction, and it needs to affect library compilation.)

Actual behavior: After `Font.default` has been changed, it persists through library recompilation. The default font isn't the same kind of special case as `tailCallOptimize`, so I'd argue that this is a bug. It must be a simple matter of a global variable somewhere that library compilation forgot to touch.

**Comment by capital-G** [comment]
Can not reproduce this behavior on 3.11.2 w/ macOS 11.

```supercollider
StaticText(nil, Rect(800, 200, 100, 20)).string_("test").front;
```

![image](https://user-images.githubusercontent.com/8267062/128252720-24791992-9e4b-4f7f-ad5a-a9c1569bfba0.png)

```supercollider
// This displays with serifs
Font.default = Font("Times New Roman", 12);
StaticText(nil, Rect(800, 200, 100, 20)).string_("test").front;
```

![image](https://user-images.githubusercontent.com/8267062/128252859-4c879c7e-5bf5-4fbb-b06a-daa4418d1d18.png)

 ```supercollider
thisProcess.recompile;

// This *still* displays with serifs
StaticText(nil, Rect(800, 200, 100, 20)).string_("test").front;
```

![image](https://user-images.githubusercontent.com/8267062/128252964-a5507eb4-43c0-4e98-a2e8-743ea830f804.png)

**Comment by adcxyz** [comment]
still works as expected in macOS 13.6.6, SC 3.13.0. 
@jamshark70, does this still happen in linux?

**Comment by JordanHendersonMusic** [comment]
I can't reproduce on linux, goes back to the original font after recompiling.

**Comment by jamshark70** [comment]
> still works as expected in macOS 13.6.6, SC 3.13.0.
> @jamshark70, does this still happen in linux?

Yes, it does. I just retested and the default font doesn't reset.

Here, for example, is a ddwSnippets dialog, immediately after booting the interpreter.

![sc-font-default](https://github.com/supercollider/supercollider/assets/318301/900812b3-e9f3-4c1f-944e-c647e774be24)

And here it is, after setting `Font.default = Font("Liberation Sans", 24);` and recompiling the class library. I'm including part of the post window to prove that sclang did only its initialization sequence -- there's no `-> result` after recompile. If I set the font after recompile, you'd see `-> Font` in the post window -- so the font difference can *only* have been inherited from before recompiling.

![sc-font-not-reset](https://github.com/supercollider/supercollider/assets/318301/c4782f72-dfb5-4eb3-9311-dcaec5f0af7e)

Since it seems to be only me, I guess I'll have to work around it by explicitly setting the font in my startup.scd.

It seems to me that this is a little bit similar to uninitialized memory, which is why I brought it up as a bug -- in some environments, something is being left to chance.

**Comment by telephon** [comment]
Yes, it would be good if more people could test it, maybe there is a pattern.

**Comment by JordanHendersonMusic** [comment]
I'm using KDE, which is Qt. Perhaps this has something to do with it?
